content/api/node.md

```diff
@@ -1,7 +1,301 @@
+
+* `.run(callback)`
+* `.watch(watchOptions, handler)`
+
+## Run
+
+Calling the `run` method on the `Compiler` intance is much like the quick run method mentioned above:
+
+``` js-with-links
+const webpack = require("webpack");
+
+const compiler = webpack({
+  // [Configuration Object](/configuration/)
+});
+
+compiler.run((err, [stats](#stats-object)) => {
+  // ...
+});
+```
+
+## Watching
+
+Calling the `watch` method, triggers the webpack runner, but then watches for changes (much like CLI: `webpack --watch`), as soon as webpack detects a change, runs again. Returns an instance of `Watching`.
+
+``` js-with-links
+watch(watchOptions, callback)
+```
+
+``` js-with-links-with-details
+const webpack = require("webpack");
+
+const compiler = webpack({
+  // [Configuration Object](/configuration/)
+});
+
+const watching = compiler.watch({
+  <details><summary>/* [watchOptions](/configuration/watch/#watchoptions) */</summary>
+  aggregateTimeout: 300,
+  poll: undefined
+  </details>
+}, (err, [stats](#stats-object)) => {
+  // Print watch/build result here...
+  console.log(stats);
+});
+```
+
+`Watching` options are [covered in detail here](/configuration/watch/#watchoptions).
+
+### Close `Watching`
+
+The `watch` method returns a `Watching` instance that exposes `.close(callback)` method. Calling this method will end watching:
+
+``` js
+watching.close(() => {
+  console.log("Watching Ended.");
+});
+```
+
+T> It’s not allowed to watch or run again before the existing watcher has been closed or invalidated.
+
+### Invalidate `Watching`
+
+Manually invalidate the current compiling round, but don’t stop watching.
+
+``` js
+watching.invalidate(() => {
+  console.warn("Invalidated.");
+});
+```
+
+## Stats Object
+
+The `stats` object that is passed as a second argument of the [`webpack()`](#webpack-) callback, is a good source of information about the code compilation process. It includes:
+
+- Errors and Warnings (if any)
+- Timings
+- Module and Chunk information
+- etc.
+
+The [webpack CLI](/api/cli) uses this information to display a nicely formatted output in your console.
+
+This object exposes these methods:
+
+### `stats.hasErrors()`
+
+Can be used to check if there were errors while compiling. Returns `true` or `false`.
+
+### `stats.hasWarnings()`
+
+Can be used to check if there were warnings while compiling. Returns `true` or `false`.
+
+### `stats.toJSON(options)`
+
+Returns compilation information as a JSON object. `options` can be either a string to specify the granularity level or an object of options in which you can customize the output completely. E.g:
+
+``` js
+stats.toJSON("minimal");
+```
+
+| Option | Alternative | Description |
```

Comment:
These options are also documented on the [stats page](https://github.com/webpack/webpack.js.org/blob/develop/content/configuration/stats.md), albeit a lot shorter. I think it would make more sense to move this table to the stats page instead, so non-API users also get this information.